pkg/oracle/request.go

```diff
@@ -22,6 +22,28 @@ func (o *Oracle) RemoveRequests(ids []uint64) {
 
 // AddRequests saves all requests in-fly for further processing.
 func (o *Oracle) AddRequests(reqs map[uint64]*state.OracleRequest) {
+	if len(reqs) == 0 {
+		return
+	}
+
+	// We care only about last request map, so if there is
+	// an unprocessed map, we can remove it safely.
```

Comment:
Why is that? You're collecting these requests per-block and doing `AddRequests()` in `OnPersistEnd()`, so the next batch of requests won't contain requests from the previous one and by discarding unprocessed map here you're effectively discarding these requests.

Reply:
This was fixed, forgot to remove comment.